generators/php/model/src/union/UnionGenerator.ts

```diff
@@ -0,0 +1,1329 @@
+            );
+        }
+
+        parameters.push(
+            php.parameter({
+                name: "_unknown",
+                type: php.Type.mixed()
+            })
+        );
+
+        const body = php.codeblock((writer) => {
+            const constructorArgs: php.Map.Entry[] = [];
+
+            for (const property of this.unionTypeDeclaration.baseProperties) {
+                constructorArgs.push({
+                    key: php.codeblock(`'${property.name.wireValue}'`),
+                    value: php.codeblock(this.context.getVariableName(property.name.name))
+                });
+            }
+
+            constructorArgs.push({
+                key: php.codeblock(`'${this.unionTypeDeclaration.discriminant.wireValue}'`),
+                value: php.codeblock("'_unknown'")
+            });
+            constructorArgs.push({
+                key: php.codeblock("'value'"),
+                value: php.codeblock("$_unknown")
+            });
+
+            const constructorCall = php.instantiateClass({
+                classReference: this.classReference,
+                arguments_: [php.map({ entries: constructorArgs, multiline: true })]
+            });
+
+            writer.write("return ");
+            writer.writeNodeStatement(constructorCall);
+        });
+
+        return php.method({
+            name: "_unknown",
+            access: "public",
+            parameters,
+            return_: php.Type.reference(this.classReference),
+            body,
+            static_: true
+        });
+    }
+
+    private staticConstructor(variant: SingleUnionType): php.Method {
+        return php.method({
+            name: this.context.getPropertyName(variant.discriminantValue.name),
+            access: "public",
+            parameters: this.getStaticConstructorParameters(variant),
+            return_: php.Type.reference(this.classReference),
+            body: this.getStaticConstructorBody(variant),
+            static_: true
+        });
+    }
+
+    private getStaticConstructorParameters(variant: SingleUnionType): php.Parameter[] {
+        const parameters: php.Parameter[] = [];
+
+        for (const property of this.unionTypeDeclaration.baseProperties) {
+            parameters.push(
+                php.parameter({
+                    name: this.context.getPropertyName(property.name.name),
+                    type: this.context.phpTypeMapper.convert({ reference: property.valueType })
+                })
+            );
+        }
+
+        switch (variant.shape.propertiesType) {
+            case "samePropertiesAsObject":
+            case "singleProperty":
+                parameters.push(
+                    php.parameter({
+                        name: this.context.getPropertyName(variant.discriminantValue.name),
+                        type: this.getReturnType(variant)
+                    })
+                );
+                break;
+            case "noProperties":
+            default:
+        }
```

Comment:
Let's use `assertNever` in the `default` case (like [this](https://github.com/fern-api/fern/blob/69b7daeadf4c29442b0bb49f18f48daab0df8346/generators/php/sdk/src/endpoint/http/HttpEndpointGenerator.ts#L285)). That way if we ever update the number of cases, we won't get a silent failure here.

```suggestion
        switch (variant.shape.propertiesType) {
            case "samePropertiesAsObject":
            case "singleProperty":
                parameters.push(
                    php.parameter({
                        name: this.context.getPropertyName(variant.discriminantValue.name),
                        type: this.getReturnType(variant)
                    })
                );
                break;
            case "noProperties":
            default:
                assertNever(variant.shape);
        }
```